### PR TITLE
Fix issue calling provision without filter

### PR DESCRIPTION
**Pull request by caveman-dick**

### CODE REVIEW

**Comment by fabriziopandini** [comment]
@caveman-dick thanks for your PR! I will have a look at it ASAP.

**Comment by caveman-dick** [comment]
No probs. Just a minor issue I found. Sorry I didn't add a comment, it was a little late when I submitted it! :) I'm using ansible_local to provision my servers so I currently run:

```
vagrant provision --provision-with ansible_local
```

If I need to make any changes to the ansible play. The `--provision-with ansible_local` was being picked up as a filter and causing the inventory generation to fail and not add the nodes to the groups. It's not very sophisticated but worked for me! :)

File: lib/vagrant/compose/declarative/cluster.rb [hunk 1]
```diff
@@ -17,18 +17,10 @@ class Cluster
...
-        attr_reader   :multimachine_filter
-
         # The root path for ansible playbook; it is used as a base path for computing ansible_group_vars and ansible_host_vars
         # It defaults to current directory/provisioning
         attr_accessor :ansible_playbook_path
 
-        # Costruttore di una istanza di cluster.
-        def initialize()
-          @multimachine_filter = ""
-        end
```
Comment (fabriziopandini) [inline lib/vagrant/compose/declarative/cluster.rb:30]: ok, thanks, @multimachine_filter here is not necessary.
multimachine_filter functionalities is now implemented in config.rb

File: lib/vagrant/compose/programmatic/cluster.rb [hunk 2]
```diff
@@ -39,7 +39,6 @@ def initialize(name)
           @ansible_context_vars  = {}
           @ansible_group_vars    = {}
           @ansible_host_vars     = {}
-          @multimachine_filter   = ""
```
Comment (fabriziopandini) [inline lib/vagrant/compose/programmatic/cluster.rb:42]: ok, thanks, @multimachine_filter here is not necessary.
multimachine_filter functionalities is now implemented in config.rb

File: lib/vagrant/compose/declarative/cluster.rb (hunk 1 above)
Comment (caveman-dick) [reply lib/vagrant/compose/declarative/cluster.rb:30]: Yep noticed that so that's why I removed it! :)

File: lib/vagrant/compose/config.rb [hunk 3]
```diff
@@ -112,7 +112,7 @@ def debug(verbose = false)
             puts "     group_index    #{node.group_index}"
           end
 
-          filter = " (NB. filtered by #{@cluster.multimachine_filter})" if not @multimachine_filter.empty?
+          filter = " (NB. filtered by #{@multimachine_filter})" if not @multimachine_filter.empty?
```
Comment (fabriziopandini) [inline lib/vagrant/compose/config.rb:115]: ok

File: lib/vagrant/compose/config.rb [hunk 4]
```diff
@@ -22,7 +22,7 @@ def initialize
         @cluster = nil
     		@nodes = {}
         @ansible_groups ={}
-        @multimachine_filter = ((['up', 'provision'].include? ARGV[0]) && ARGV.length > 1) ? ARGV.drop(1) : [] # detect if running vagrant up/provision MACHINE
+        @multimachine_filter = ((['up', 'provision'].include? ARGV[0]) && ARGV.length > 1 && !ARGV[1].start_with?('-')) ? ARGV.drop(1) : [] # detect if running vagrant up/provision MACHINE
```
Comment (fabriziopandini) [inline lib/vagrant/compose/config.rb:25]: Ok, but I will pick up from here to improve it a little bit (e.g. handle use cases with both options and vm filters, make tolerant to order of options a,d filters etc. etc.)

**Comment by fabriziopandini** [comment]
included in v.0.7.3. 
you can run `vagrant plugin update`to get the new release

**Comment by caveman-dick** [comment]
Thanks! 👍

**Pull request merged**